oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,175 @@
+
+.. contents::
+   :local:
+   :depth: 3
+
+
+Abstract
+========
+
+Wherever possible, The edX organization will provide and manage Docker images as the medium for packaging operating dependencies of Open edX applications rather than providing Ansible playbooks.
+
+The configuration for running applications in different environments will be done through a config file per unique [application, environment] combination. How those files are generated and managed is the responsibility of each operator.
+
+All standard operations for running and managing any Open edX application will be documented. How those procedures are executed is the responsibility of each operator.
+
+Wherever possible, operating concerns will live within the codebase they are concerned with. Thus it should be expected that all applications will provide the following:
+
+* A Dockerfile from which the container image can be built.
+* A documented settings file with production-ready defaults.
+* A documented settings file with all of the configuration values that must be defined by the operator.
+* An operations manual documenting standard operations required to run and maintain the application.
+
+Context
+=======
+
+Back when edx-platform was the only code running, the Ansible playbooks in the configuration repo were manageable, though still not necessarily easy to understand. As edX has invested in microservices, the configuration repo has grown to support the configuration of all of them. As patterns emerged shared concerns were pulled out into unversioned shared roles making it hard to change without impacting all apps or wiring configurable toggles throughout much of the shared space.
+
+The end result is that the configuration repo has grown into a sprawling codebase that attempts to be everything for everyone trying to configure servers to run Open edX installations. There are a wide variety of roles, plays, and utility scripts with branching paths within each to support being executed against different base operating systems and many various configurable operational needs. None of this is helped by the fact that edX.org operates at a scale that most installations do not. With a mix of edX.org specific code, branching paths that are not regularly tested, and the high amount of required understanding around how all the pieces interact to change any of it, this approach to operating is a clear source of friction for developing and running the platform. Nothing proves this point more than the fact that the edx-platform README now explicitly warns users to leave it to the professionals.
+
+    Installing and running an Open edX instance is not simple. We strongly recommend that you use a service provider to run the software for you. 
+
+The proposal below outlines how we can create a cleaner, more intuitive interface for operating Open edX and in doing so help both edX.org and the Open edX community at large achieve better outcome at a faster pace. In brief it is:
```

Comment:
This OEP does a really good job describing how configuration should be done on each app and sets great standards for building images, but it doesn't cover the "how to put things together" like the current `configuration` repo implicitly does.

For example, when running `openedx_native.yml`, the playbooks set up, on a single instance, all the services needed to run a full Open edX deployment:
- LMS
- Studio
- Workers
- IDAs
- Other support services
- Handles setting up LMS/Studio and wiring between them and the IDAs

I know that this OEP doesn't cover `deployment` because there are a lot of different ways to do that, but the requirements for a Open edX deployment are not described anywhere (except implicitly in the `configuration` repo).
So the `configuration` repo isn't the best reference, but it's currently all we have for deploying from scratch.

Can we add a section on this OEP (or create a follow up OEP), to better document and map how the platform is supposed to work? What is each piece, and what's required for a "basic" installation if starting from scratch using the docker images?